Первая попытка

AleksandrRegulus opened the pull request:
Прога готова. Очевидно корявая, но работает.

AleksandrRegulus committed:
Написал

AleksandrRegulus committed:
Написал

Anna-Malinina commented on src/main/kotlin/MenuArchives.kt:
```diff
@@ -0,0 +1,23 @@
+        val list = currentChoice.listArchives
+        list.forEach {
+            menuList[++i] = MenuItem("Архив '${it.archiveName}'") {
+                currentChoice.numberListMenu = 2
```
Голые цифры -  [магические числа](https://ru.wikipedia.org/wiki/%D0%9C%D0%B0%D0%B3%D0%B8%D1%87%D0%B5%D1%81%D0%BA%D0%BE%D0%B5_%D1%87%D0%B8%D1%81%D0%BB%D0%BE_(%D0%BF%D1%80%D0%BE%D0%B3%D1%80%D0%B0%D0%BC%D0%BC%D0%B8%D1%80%D0%BE%D0%B2%D0%B0%D0%BD%D0%B8%D0%B5)). Сложно разобраться в коде, когда приходится догадываться, что все это значит) Всегда лучше выносить такие в константы)

Anna-Malinina commented on src/main/kotlin/UserInput.kt:
```diff
@@ -0,0 +1,21 @@
+    private val input = Scanner(System.`in`)
+
+    fun getUserAnswerMenu(): Int? {
+        val answer: Int? = if (input.hasNextInt()) input.nextInt() else null
```
Можно еще просто использовать метод toIntOrNull

AleksandrRegulus committed:
Убрал "голые цифры" заменил на enum класс
Воспользовался методом toIntOrNull
Переделал часть где создается меню: вместо кучи переменных создал массив и вместо when вызов функции от элемента массива.
Ну и в целом причесал код.

Долго думал над замечанием про повторение кода:
MenuCreateArchive и MenuCreateNote очень и очень похожи, можно вынести повторяющийся код)

Но не придумал как упростить эти 2 класса.
1.Они при всей похожести разные, они формируют меню с разными названиями и разными лямбдами. т.е. текст похож, но повторений нет
2.В теории эти меню могут обрасти новыми пунктами и тогда станут различаться еще сильнее. И тогда тем более нельзя их объединять например, т.к. это разные экраны для пользователя и разные объекты для разработчика.

AleksandrRegulus committed:
Kotlin-Module-Project.drawio

AleksandrRegulus committed:
Kotlin-Module-Project.drawio

AleksandrRegulus committed:
Kotlin-Module-Project.drawio

AleksandrRegulus committed:
Вынес повторяющийся код из MenuCreateArchive и MenuCreateNote. Стало немного симпатичнее :)
Спасибо за подробную рекоммендацию!

AleksandrRegulus committed:
Merge remote-tracking branch 'origin/dev' into dev
